src/identity/identity.ts

```diff
@@ -80,4 +93,73 @@ export default class Identity {
       this.restorer = Identity.DEFAULT_RESTORER;
     }
   }
+
+  async blindPsetWithBlindKeysGetter(
+    getBlindingKeyPair: (
+      script: Buffer
+    ) => { publicKey: Buffer; privateKey: Buffer },
+    psetBase64: string,
+    outputsToBlind: number[],
+    outputsPubKeys?: Map<number, string>,
+    inputsPrivKeys?: Map<number, string>
+  ): Promise<string> {
+    const inputsKeys: Map<number, Buffer> = new Map<number, Buffer>();
+    const outputsKeys: Map<number, Buffer> = new Map<number, Buffer>();
+
+    const pset = decodePset(psetBase64);
+    const transaction = Transaction.fromHex(psetToUnsignedHex(psetBase64));
+
+    // set the outputs map
+    for (const index of outputsToBlind) {
```

Comment:
Just thinking about this interface https://github.com/vulpemventures/documentation/blob/liquid/docs/web_extension/liquid/wallet_sign_pset.md#parameters

 If our purpose is to get the blinding pub key via  `getBlindingKeyPair`, what about asking directly for the script of the output we want to blind instead?
 
 cc @altafan

Reply:
IMO better to specify the indexes because several outputs could have the same script (even if it is not recommended). Also, from the UX point of view: do you prefer "blind outputs 2 and 3" or "let me fetch the scripts myself, and then blind outputs with script x and script y"?

Reply:
> IMO better to specify the indexes because several outputs could have the same script

If they have the same script, they will have the same blinding key isn't? As per slip77 spec. 

> let me fetch the scripts myself

Well if the user already added outputs into the PSET, it means he knows which is the address he wants to blind. The index of the output is something he can derive from rememebring the order in wich he called the `Psbt.addOutput` method.

I mean, both are ok, I like more to give the address rather than indexes, mostly to be similar to the CAL API

Reply:
> If they have the same script, they will have the same blinding key isn't? As per slip77 spec.

The only case could be if in the same tx the same script has confidential outout an another UNconfidential output.

But in this regard I would give the addresses instead, like the CAL API, so it' evident of wich outout should be blinded, based con confidential or unconfidential address

Reply:
Also hopefully the new PSET spec will encode the blind pub key, so will make this useless, so doesn't matter